backend/src/packages/articles/article.service.ts

```diff
@@ -11,9 +19,63 @@ import {
 
 class ArticleService implements IService {
   private articleRepository: ArticleRepository;
+  private openAIService: OpenAIService;
+  private genreRepository: GenreRepository;
 
-  public constructor(articleRepository: ArticleRepository) {
+  public constructor(
+    articleRepository: ArticleRepository,
+    openAIService: OpenAIService,
+    genreRepository: GenreRepository,
+  ) {
     this.articleRepository = articleRepository;
+    this.openAIService = openAIService;
+    this.genreRepository = genreRepository;
+  }
+
+  public async detectArticleGenreFromText(
+    text: string,
+  ): Promise<string | null> {
+    const generesJSON = await this.openAIService.createCompletion({
+      temperature: 0,
+      prompt: getDetectArticleGenrePrompt(text),
+    });
+
+    if (!generesJSON) {
+      return null;
+    }
+
+    const parsedGenres = safeJSONParse<string[]>(generesJSON);
+
+    return parsedGenres?.[0] ?? null;
+  }
+
+  private async getGenreIdForArticle(
+    articleText: string,
+  ): Promise<number | null> {
+    const detectedGenre = await this.detectArticleGenreFromText(articleText);
+
+    if (!detectedGenre) {
+      return null;
+    }
+
+    const detectedGenreKey = convertPlainStringToSnakeCase(detectedGenre);
```

Comment:
Can we just ask in prompt to return snake_cased key together with readable name?